python/plugins/grassprovider/Grass7Utils.py

```diff
@@ -384,6 +384,22 @@ def executeGrass(commands, feedback, outputCommands=None):
             if sys.version_info >= (3, 6):
                 kw['encoding'] = "cp{}".format(Grass7Utils.getWindowsCodePage())
 
+        def readline_with_recover(stdout):
```

Comment:
Nota: This method could also be useful for Saga and Otb plugins.

Reply:
Sounds good, you mean to apply this fix to following codes, right?
https://github.com/qgis/QGIS/blob/f261cc1f8b6549b1a5c37374a3c5aba45c92334c/python/plugins/sagaprovider/SagaUtils.py#L184

I think it may make it somewhat solid.
However, IMHO, this patch is a just workaround so it is not needed as long as no error occurs.
What do you think?

Reply:
Re-thinking this, I've changed my mind that it can be better to aplly this patch for SAGA's one in terms of preventing similar errors.

Reply:
@lbartoletti 
Thank you for a comment, I've add a patch to SAGA as well.